marklogic-data-hub-central/ui-custom/src/components/confirmation-modal/confirmation-modal.tsx

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import {Modal} from "react-bootstrap";
+import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
+import {faExclamationTriangle} from "@fortawesome/free-solid-svg-icons";
+import "./confirmation-modal.scss";
+
+type Props = {
+  isVisible: boolean;
+  toggleModal: (isVisible: boolean) => void;
+  confirmAction: () => void;
+  bodyContent: string;
+  headerContent: string;
+};
+
+const ConfirmationModal: React.FC<Props> = (props) => {
+
+  const closeModal = () => {
+    props.toggleModal(false);
+  };
+
+  const handleYesClicked = () => {
+    props.confirmAction();
+    props.toggleModal(false);
+  }
+
+  return (
+    <Modal
```

Comment:
Please wrap the content in this component in a "confirmation-modal" class, so we can isolate the CSS classes to avoid pollution in other widgets.